openspoor/mapservices/MapservicesQuery.py

```diff
@@ -140,10 +158,45 @@ def _transform_dict_to_gdf(self, data):
         map_services.prorail.nl
         :return: geopandas dataframe with geometry or pandas dataframe without
         """
+        geometry_list = [
+            shape(feature['geometry']) if feature['geometry'] is not None else None 
+            for feature in data['features']]
         attribute_list = [feature['properties'] for feature in data['features']]
-        geometry_list = [feature['geometry'] for feature in data['features']]
         if all(geometry is None for geometry in geometry_list):
             return pd.DataFrame(data=attribute_list)
-        geometry_list = [shape(geometry) if geometry is not None else None 
-                         for geometry in geometry_list]
-        return gpd.GeoDataFrame(data=attribute_list, geometry=geometry_list, crs=self.crs)
\ No newline at end of file
+        return gpd.GeoDataFrame(data=attribute_list, geometry=geometry_list, crs=self.crs)
+
+        
+class MapServicesQueryMValues(MapServicesQuery):
+    """
+    Class for getting M-values from the mapservices.prorail.nl feature servers.
+    Unfortunately these are in json format only, so the standard query needs to be adjusted for this.    
+    """
+
+    def __init__(self, url: Optional[str] = None, cache_location: Optional[Path] = None):
+        super().__init__(url, cache_location)
+        self.standard_featureserver_query = self.standard_featureserver_query.replace('f=geojson', 'f=json') + \
+                                            '&returnM=true&returnZ=false'
+
+    def _transform_dict_to_gdf(self, data: dict) -> gpd.GeoDataFrame:
+        """
+        Transform given json format data from feature servers into a geopandas
+        dataframe with given geometry.
+
+        :param data: dictionary, json format as retrieved from feature server
+        map_services.prorail.nl
+        :return: geopandas dataframe with geometry or pandas dataframe without
+        """
+        attribute_list = [feature['attributes'] for feature in data['features']]
+        if data['geometryType'] == 'esriGeometryPolyline':
+            geometry_list = [LineString([tuple(p) for p in
+                                        f['geometry']['paths'][0]])
```

Comment:
This one seems to be similar to what was there before. However this one fails for multilinestrings. then only the first linestring is returned in the result. For a layer with multilinestring see the **dienstregelpunten lijn**

Reply:
This should not be a concern for the current layers, but the functionality was extended nonetheless.
To make sure an exhaustive list of types will not be required a warning is now thrown if a user requests m values for a layer which does not contain it will now throw a warning and notimplementederror (for cases which are not implemented)
<img width="591" alt="image" src="https://github.com/user-attachments/assets/cdf4fecd-622b-4ae1-8661-a99ef7cb0006">

![image](https://github.com/user-attachments/assets/df729b35-c0d5-42e8-b10d-598489cc964c)